src/main/scala/com/softwaremill/realworld/articles/core/ArticlesRepository.scala

```diff
@@ -162,16 +158,35 @@ class ArticlesRepository(quill: Quill.Sqlite[SnakeCase]):
           _.updatedAt -> lift(now),
           _.authorId -> lift(userId)
         )
-        .returning(_.articleId)
-    )
-      .pipe(mapUniqueConstraintViolationError)
-  }
+        .returning[Int](_.articleId)
+    }
 
-  def deleteArticle(articleId: Int): Task[Long] =
-    run(queryArticle.filter(_.articleId == lift(articleId)).delete)
+    def addTags(tags: List[String], articleId: Int) =
+      quote {
+        liftQuery(tags).foreach(tag => queryTagArticle.insertValue(ArticleTagRow(tag, lift(articleId))))
+      }
 
-  def deleteFavoritesByArticleId(articleId: Int): Task[Long] =
-    run(queryFavoriteArticle.filter(_.articleId == lift(articleId)).delete)
+    transaction {
+      run(addArticle)
+        .pipe(mapUniqueConstraintViolationError)
+        .flatMap(articleId => run(addTags(tags, articleId)))
+        .unit
+    }
+  }
+
+  def deleteArticleTransaction(articleId: Int): Task[Long] = {
+    val deleteComments = queryComment.dynamic.filter(_.commentId == lift(articleId)).delete
+    val deleteFavorites = queryFavoriteArticle.dynamic.filter(_.articleId == lift(articleId)).delete
+    val deleteTags = queryTagArticle.dynamic.filter(_.articleId == lift(articleId)).delete
+    val deleteArticle = queryArticle.dynamic.filter(_.articleId == lift(articleId)).delete
+
+    transaction {
+      run(deleteComments).unit
+        .flatMap(_ => run(deleteFavorites).unit)
```

Comment:
**suggestion:** I think you don't need to `_.unit` if you're flatMapping with `_ =>`